pkg/ebpf/c/tracee.bpf.c

```diff
@@ -288,7 +288,49 @@ Copyright (C) Aqua Security inc.
 #define get_kconfig(x) CONFIG_##x
 #endif
 
+#ifdef CORE
+
 #define ARCH_HAS_SYSCALL_WRAPPER        1000U
+#define FLATMEM                         1001U
+#define	DISCONTMEM                      1002U
+#define	SPARSEMEM                       1003U
+#define	SPARSEMEM_VMEMMAP               1004U
+#define	SPARSEMEM_EXTREME               1005U
+#define	DYNAMIC_MEMORY_LAYOUT           1006U
+#define	X86_32                          1007U
+#define	X86_PAE                         1008U
+
+#else
+
+#ifndef CONFIG_ARCH_HAS_SYSCALL_WRAPPER
+#define CONFIG_ARCH_HAS_SYSCALL_WRAPPER 0
+#endif
```

Comment:
How will these be used? How are defining them here different than loading it into the `kconfig_map` BPF_HASH map?

Reply:
When using no-CORE, we have the issue that the kconfig values might not be defined.
To bypass it, I defined them in the case they are not defined with the value of 0 to be able to create a uniform behavior using no-CORE and CORE with the `get_kconfig` macro.
If you have better solution to this issue I will gladly hear :)

Reply:
Thanks @grantseltzer for checking this and @AlonZivony to keep using what we came up with. The reasoning behind this is the following:

If you ever use a kconfig relocation (done by libbpf when you declare an external symbol coming from the kconfig file) by using the following syntax: extern KCONFIG_OPTION __kconfig, then libbpf will, FOR SURE, require a kconfig file existence (/boot/kconfig-$(uname -r) OR /proc/kconfig.gz). 

@yanivagman and I decided that, in order not to be forced to have those 2 files, and being able to assume kconfig values, we would pass the values through kconfig map. With that, we can "mock" kconfig values to a container, for example, without requiring the container to have a kconfig file (which libbpf would require).

There are pros and cons with that approach, for sure, and we may revisit this in the near future. For now, it was the right thing to do (Alon using the same approach), IMO.

Thanks!